src/core/entities/MRTextAreaEntity.js

```diff
@@ -56,183 +41,124 @@ export class MRTextAreaEntity extends MRTextInputEntity {
+        this.hiddenInput.maxlength = this.getAttribute('maxlength') ?? undefined;
         // wrap: Controls how text is wrapped in the textarea, with values like soft and hard affecting form submission.
-        this.hiddenInput.wrap = this.getAttribute('wrap') ?? this.defaults.wrap;
+        this.hiddenInput.wrap = this.getAttribute('wrap') ?? undefined;
         // overflowwrap : Controls how wrap breaks, at whitespace characters or in the middle of words.
-        this.textObj.overflowWrap = this.getAttribute('overflowWrap') ?? this.defaults.overflowWrap;
+        this.hiddenInput.overflowWrap = this.getAttribute('overflowWrap') ?? undefined;
         // whitespace : Controls if text wraps with the overflowWrap feature or not.
-        this.textObj.whiteSpace = this.getAttribute('whitespace') ?? this.defaults.whiteSpace;
-    }
-
-    /**
-     * Overrides the connected method to include setup for handling multiline text.
-     * @function
-     * @description (async) sets up the textObject of the text item.
-     */
-    async connected() {
-        await super.connected();
+        this.hiddenInput.whiteSpace = this.getAttribute('whitespace') ?? undefined;
     }
 
     /**
      *
      */
     updateTextDisplay() {
-        // Determine the maximum number of characters per line based on renderable area (example given)
-        const maxCharsPerLine = 50; // This should be dynamically calculated
-
-        const lines = this.hiddenInput.value.split('\n').map((line) => {
-            // Truncate or split lines here based on maxCharsPerLine if implementing horizontal scrolling
-            return line.substring(0, maxCharsPerLine);
-        });
-
-        // Existing logic to determine visibleLines based on scrollOffset and maxVisibleLines
-        const visibleLines = lines.slice(this.scrollOffset, this.scrollOffset + this.maxVisibleLines);
-        const visibleText = visibleLines.join('\n');
+        // XXX - add scrolling logic in here for areas where text is greater than
+        // the width/domain the user creates visually
 
-        this.textObj.text = visibleText;
-        // console.log('text updated: ', this.textObj.text);
-
-        // Logic to adjust scrollOffset for new input, ensuring the latest text is visible
-        if (lines.length > this.maxVisibleLines && this.hiddenInput === document.activeElement) {
-            this.scrollOffset = Math.max(0, lines.length - this.maxVisibleLines);
-        }
-
-        this.updateCursorPosition();
+        this.textObj.text = this.hiddenInput.value;
     }
 
     /**
-     * Handles keydown events for scrolling and cursor navigation.
+     * Handles keydown events for scrolling and cursor navigation. Note
+     * that this is different than an input event which for our purposes,
+     * handles the non-navigation key-presses.
      * @param {event} event - the keydown event
      */
     handleKeydown(event) {
```

Comment:
This really shouldn't be necessary if we're using the `hiddenInput`, right? you can get the selection (cursor position) directly from that.

Reply:
hidden input doesnt handle the up/down keypresses as we would expect in a larger space - especially for textarea, it only treats them as left and right - hence the separation (see the comments in the `setupEventListeners` function)

that's something i'm going to simplify as part of the next pr given the lingering issue for '\n', but unfortunately for now both are still needed

Reply:
I was using the keyboard events in the first version (prefunding), and it had quite a few issues, one being accessibility, as it doesn't launch the virtual keyboard.

input should work, I think you'd need to change it being `document.createElement('input')` to `document.createElement('textarea')` and that would give you the rows info.

Reply:
Also events have a nasty habit of being caught out of order in some cases.